File: Data Structures & Algorithms/eating-bananas/submission-1.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        int result = piles.Max();
        int left = 1;
        int right = piles.Max();
        while (left <= right)
        {
            int mid = (left + right) / 2;
            long thisResult = 0;
            foreach (int num in piles)
            {
                thisResult += (int)Math.Ceiling((double)num / mid);
            }
            if (thisResult <= h)
            {
                result = Math.Min(result, mid);
                right = mid - 1;
            }
            else
            {
                left = mid + 1;
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        if len(nums) < 3:
            return []
        elif len(nums) == 3 and nums[0] + nums[1] + nums[2] != 0:
            return []
        elif len(nums) == 3 and nums[0] + nums[1] + nums[2] == 0:
            return [[nums[0], nums[1], nums[2]]]

        result = []
        sortedNums = sorted(nums)
        for index, num in enumerate(sortedNums):
            if num > 0:
                break
            if index > 0 and num == sortedNums[index - 1]:
                continue

            left, right = index + 1, len(sortedNums) - 1
            while left < right:
                thisResult = num + sortedNums[left] + sortedNums[right]
                if thisResult < 0:
                    left += 1
                elif thisResult > 0:
                    right -= 1
                else:
                    result.append([num, sortedNums[left], sortedNums[right]])
                    left += 1
                    right -= 1
                    while left < right and sortedNums[left] == sortedNums[left - 1]:
                        left += 1
        return result